File: 0690-employee-importance/0690-employee-importance.rb
=begin
# Definition for Employee.
class Employee
    attr_accessor :id, :importance, :subordinates
    def initialize( id, importance, subordinates)
        @id = id
        @importance = importance
        @subordinates = subordinates
    end
end
=end

# @param {Employee} employees
# @param {Integer} id
# @return {Integer}
def get_importance(employees, id)
  result=0
  p manager = employees.find { |emp| emp.id == id }
  result = manager.importance
  manager.subordinates.each do |sub|
    result += get_importance(employees, sub)
  end
  result
end

File: 0690-employee-importance/0690-employee-importance.py
"""
# Definition for Employee.
class Employee:
    def __init__(self, id: int, importance: int, subordinates: List[int]):
        self.id = id
        self.importance = importance
        self.subordinates = subordinates
"""

class Solution:
    def getImportance(self, employees: List['Employee'], id: int) -> int:
        return (c:=next(i for i in employees if  i.id==id)).importance+sum(self.getImportance(employees,i) for i in c.subordinates)      